sympy/functions/special/hyper.py

```diff
@@ -251,14 +251,8 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         terms = []
 
         for i in range(n):
-            num = 1
-            den = 1
-            for a in ap:
-                num *= RisingFactorial(a, i)
-
-            for b in bq:
-                den *= RisingFactorial(b, i)
-
+            num = Mul(*[RisingFactorial(a, i) for a in ap])
+            den = Mul(*[RisingFactorial(b, i) for b in bq])
```

Comment:
This is such a common pattern in the codebase that maybe there should be a convenience function for doing this like:
```python
def _Mul(args):
    return Mul(*args)
```
Or perhaps it can be called `sprod` for "symbolic product".

Then it can just be
```python
num = _Mul(RisingFactorial(a, i) for a in ap)
num = sprod(RisingFactorial(a, i) for a in ap)
```

Reply:
Is that really significantly better?

(Now I use a list since that seems faster than lazy evaluation, but apart from that?)

Reply:
The earlier comment was more from a "rewriting loops"-aspect.

Reply:
The `prod` function is `O(N**2)` on symbolic expressions because `Mul.flatten` runs N times.

Reply:
The difference is pretty dramatic - for 1024 random symbols `Mul(*syms)` takes 0.025 seconds while `prod` takes 22.3. There is about a 10% slowdown for the int case if we do this:
```python
def prod(f, start=1):
    """return an int if all elements are ints else a SymPy expression"""
    from sympy.utilities.iterables import sift
    n, s = sift(f, lambda x: type(x) is int, binary=True)
    if start != 1:
        if type(start) is int:
            n.append(start)
	else:
	    s.append(start)
    if n:
        from math import prod as p
        return p(n) if not s else p(n)*Mul(*s)
    return Mul(*s)
```
But it seems better to me to just update internal uses of `prod` on ints to use `math.prod` and use `Mul(*factors)` or `sprod` (tba) for others and deprecate `mul.prod`.

Reply:
We do mostly already use `Mul(*factors)`. I don't think `mul.prod` needs to be deprecated but we could change its docstring to recommend `math.prod` (and have it call `math.prod` internally).